server/middleware/searchHandler.js

```diff
@@ -47,10 +50,9 @@ async function handleSearch(req, res, next) {
             .sort(sortObjectByProp(workstackCase => workstackCase.caseReference))
             .map(bindDisplayElements(fromStaticList)));
         const { workstackTypeColumns } = await req.listService.fetch('S_SYSTEM_CONFIGURATION');
-        // using DEFAULT columns for search
-        const workstackTypeForSearch = workstackTypeColumns.find(
-            item => item.workstackType === 'DEFAULT'
-        );
+
+        const getSearchResultWorkstackTypeColumns = getWorkstackTypeColumns('SEARCH_RESULTS');
+        const workstackTypeForSearch = getSearchResultWorkstackTypeColumns(workstackTypeColumns, request);
```

Comment:
Since the method set on line 54 is only used once we could just amalgamate it into one so like:

```
workstackTypeForSearch = extractWorkstackTypeColumns('SEARCH_RESULTS')(workstackTypeColumns, request);
```

Reply:
I have made this change